Parser.java

```diff
@@ -2,41 +2,55 @@
 import java.io.FileInputStream;
 import java.io.FileOutputStream;
 import java.io.IOException;
-/**
- * This class is thread safe.
- */
+
+
 public class Parser {
   private File file;
-  public synchronized void setFile(File f) {
+  private static int UNICODE_IDENTIFIER = 0x80;
+
+  public final synchronized void setFile(File f) {
     file = f;
   }
-  public synchronized File getFile() {
+
+  public final synchronized File getFile() {
     return file;
   }
-  public String getContent() throws IOException {
-    FileInputStream i = new FileInputStream(file);
-    String output = "";
-    int data;
-    while ((data = i.read()) > 0) {
-      output += (char) data;
```

Comment:
The methods below (getContent()and getContentWithoutUnicode()) can be removed and instead a single method like this would be better:

  public final String getContent(File file, String encoding) throws IOException {
    synchronized (file) {
      StringBuilder builder = new StringBuilder();
      BufferedReader in =
          new BufferedReader(new InputStreamReader(new FileInputStream(file), encoding));

      String str;

      while ((str = in.readLine()) != null) {
        builder.append(str);
      }
      return builder.toString();
    }
  }

While saving a String to a File, again we should give an encoding as parameter.

Reading/writing should always be done with encoding.